history/management/commands/run_bot.py

```diff
@@ -272,44 +296,158 @@ def accepted(message,opponentname,gamename):
+                    if len(strings) > 5:
+                        message.send("That message is too long; I can't quite parse it :confused:")
+                        return
+                    tags = ""
+                    for x in range(2, len(strings)):
+                        tags += "#" + strings[x] + " "
+                    tags += "#" + gamename
+                    message.reply("Did you mean `won {} {} {}`? I'm a bit confused...".format(strings[0], strings[1], tags))
+                    return 
+                # Ignore 'games' that are actually tags
+                if "#" in gamename:
+                    return
 
             #setup
             sender = "@" + message.channel._client.users[message.body['user']][u'name']
             opponentname = _get_user_username(message,opponentname)
 
             #body
-            Game.objects.create(winner=sender,loser=opponentname,gamename=gamename,created_on=datetime.now(),modified_on=datetime.now())
+            newgame = Game.objects.create(winner=sender,loser=opponentname,gamename=gamename,created_on=datetime.now(),modified_on=datetime.now())
+
+            # Create tags for game
+            if len(arg) > 3:
+                for x in range(3, len(arg)):
+                    Tag.objects.create(tag=arg[x], game=newgame)
 
             #send response
             gifurl = get_gif('winorloss')
             message.send("#win recorded \n {}".format(gifurl))
-
-        @listen_to('^lost (.*) (.*)',re.IGNORECASE)
-        @listen_to('^loss (.*) (.*)',re.IGNORECASE)
-        def loss(message,opponentname,gamename):
+            if gamename == "chess":
+                elo_rankings = _get_elo(gamename)
+                message.send(":arrow_up: {0}'s new elo: {1}\n:arrow_down: {2}'s new elo: {3}\n".format(sender, elo_rankings[sender], opponentname, elo_rankings[opponentname]))
+
+
+        @listen_to('lost (.*) (.*)$',re.IGNORECASE)
+        @listen_to('lost (.*) (.*) #(.*)$',re.IGNORECASE)        
+        @listen_to('lost (.*) (.*) #(.*) #(.*)$',re.IGNORECASE)
+        @listen_to('lost (.*) (.*) #(.*) #(.*) #(.*)$',re.IGNORECASE)
+        @listen_to('lost (.*) (.*) #(.*) #(.*) #(.*) #(.*)$',re.IGNORECASE)
+        @listen_to('lost (.*) (.*) #(.*) #(.*) #(.*) #(.*) #(.*)$',re.IGNORECASE)
+        @listen_to('loss (.*) (.*)$',re.IGNORECASE)
+        @listen_to('loss (.*) (.*) #(.*)$',re.IGNORECASE)        
+        @listen_to('loss (.*) (.*) #(.*) #(.*)$',re.IGNORECASE)
+        @listen_to('loss (.*) (.*) #(.*) #(.*) #(.*)$',re.IGNORECASE)
+        @listen_to('loss (.*) (.*) #(.*) #(.*) #(.*) #(.*)$',re.IGNORECASE)
+        @listen_to('loss (.*) (.*) #(.*) #(.*) #(.*) #(.*) #(.*)$',re.IGNORECASE)
+        def loss(*arg):
+            message = arg[0]
+            opponentname = arg[1]
+            gamename = arg[2]
             #input sanitization
             gamename = gamename.strip()
+            if " " in opponentname:
+                if "#" not in gamename:
```

Comment:
any objections to refactoring this (and lines 318 - 332 above) into a helper function?  im allergic to copy paste code :-P